src/settings/LanguagePicker.js

```diff
@@ -15,23 +15,53 @@ const styles = StyleSheet.create({
 type Props = {|
   value: string,
   onValueChange: (locale: string) => void,
+  filter: string,
 |};
 
-export default class LanguagePicker extends PureComponent<Props> {
+type State = {|
+  data: Array<*>,
+|};
+
+export default class LanguagePicker extends PureComponent<Props, State> {
+  state = {
+    data: languages,
+  };
+
+  handleFilterChange = (text: string) => {
+    const newData = languages.filter(item => {
+      const itemData = `${item.name.toUpperCase()} ${item.nativeName.toUpperCase()}`;
+      const textData = text.toUpperCase();
+      return itemData.indexOf(textData) > -1;
```

Comment:
Use `string.prototype.includes` to make this a little bit clear.

Reply:
Updated!